src/FormBuilder/resources/js/form-builder/src/blocks/fields/amount/inspector/index.tsx

```diff
@@ -121,32 +121,45 @@ const Inspector = ({attributes, setAttributes}) => {
 
     const [donationLevels, setDonationLevels] = useState<OptionProps[]>(
         levels.map((level) => ({
+            id: String(Math.floor(Math.random() * 1000000)),
             label: formatCurrencyAmount(level.toString()),
             value: level.toString(),
             checked: defaultLevel === level,
         }))
     );
 
-    const handleLevelsChange = (options: OptionProps[]) => {
-        if (options.length > 1 && options[options.length - 1].value === '') {
-            const values = options.filter((option) => Number(option.value) > 0).map((option) => Number(option.value));
-            options[options.length - 1].value = String(2 * Math.max(...values));
-        } else if (options.length === 1 && options[0].value === '') {
-            options[0].value = '10';
-        }
+    const handleLevelAdded = () => {
+        const newLevelValue = levels.length ? String(Math.max(...levels) * 2) : '10';
+        const newLevel = {
+            id: String(Math.floor(Math.random() * 1000000)),
+            label: formatCurrencyAmount(newLevelValue),
+            value: newLevelValue,
+            checked: false,
+        };
 
-        const checkedLevel = options.filter((option) => option.checked);
+        setDonationLevels([...donationLevels, newLevel]);
+        setAttributes({levels: [...levels, Number(newLevelValue)]});
```

Comment:
If we delete all levels and then add the first one again, this function will not set it as default. Beyond not having a default item set, the drag-and-drop reorder will also not work.

To prevent that we need to add here something like that:

```
if (levels.length === 0) {
	setDonationLevels([
		{
			id: String(Math.floor(Math.random() * 1000000)),
			label: formatCurrencyAmount(newLevelValue.toString()),
			value: newLevelValue.toString(),
			checked: true,
		},
	]);

	setAttributes({defaultLevel: Number(newLevelValue)});
}
```